Sources/Locker/Internal/LockerHelpers.swift

```diff
@@ -0,0 +1,184 @@
+        // storing that nil and comparing it to nil will result as `isEqual` NO
+        // even data is not actually changed.
+        let biometricsDeactivated: Bool   = (oldDomainState == nil)  || (newDomainState == nil)
+        let biometricSettingsDidChange = oldDomainState?.elementsEqual(newDomainState!) ?? false
+        if (biometricsDeactivated && biometricSettingsDidChange) {
+            biometricsSettingsChangedStatus = true
+
+            LockerHelpers.setLAPolicyDomainState(with: newDomainState)
+        }
+
+        return biometricsSettingsChangedStatus
+    }
+
+    static var configureBiometricsAuthentication: BiometricsType {
+        let canUse = LAContext().canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil)
+
+        if canUse && LockerHelpers.canUseAuthenticationWithFaceID {
+            return .faceID
+        } else if canUse {
+            return .touchID
+        } else {
+            return .none
+        }
+    }
+
+    static var keyLAPolicyDomainState: String {
+        String(format: "%@_UserDefaultsLAPolicyDomainState", locale: nil, LockerHelpers.kBundleIdentifier)
+    }
+
+    static var canUseAuthenticationWithFaceID: Bool {
+        get {
+            let context = LAContext()
+            var error: NSError? = nil
+            if #available(iOS 11.0, *) {
+                if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error)
+                    && context.responds(to: #selector(getter: context.biometryType)) {
+                    if context.biometryType == .faceID {
+                        return true
+                    }
+                }
+            }
+            return false;
+        }
+    }
+
+    static var deviceCode: String {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children.reduce("") { identifier, element in
+            guard let value = element.value as? Int8, value != 0 else { return identifier }
+            return identifier + String(UnicodeScalar(UInt8(value)))
+        }
+        return identifier
+    }
+
+    static func deviceSupportsAuthenticationWithFaceID(_ completion: @escaping ((Bool) -> Void)) {
+        if LockerHelpers.canUseAuthenticationWithFaceID {
+            completion(true)
+        } else {
+            URLSession.shared.dataTask(with: deviceList.api) { data, _, error in
+                guard let data = data, error == nil else {
+                    completion(false)
+                    return
+                }
+                do {
+                    let deviceList = try JSONDecoder().decode(DeviceResponse.self, from: data).devices
+                    completion(deviceList.contains(LockerHelpers.deviceCode))
+                } catch {
+                    completion(false)
+                }
+            }.resume()
```

Comment:
Should be moved to the DeviceList. The amount of the API calls lib is making should be reduced to a minimum, if possible we should avoid making the API call at all.

The idea is to have an initial list of devices stored locally (inside the library). If the current device is not in the local list, only then we should make an API call. Once the API call is finished, new values are stored locally. Next time an app checks the biometrics state, the new device should be in that list, and an API call should not be triggered.

Let's just discuss if we will introduce the TouchID list in combination with the FaceID. That way, we should change the keychain logic a little bit.